src/test/datascience/widgets/thirdpartyWidgets.vscode.common.test.ts

```diff
@@ -32,199 +34,206 @@ import {
     executeCellAndWaitForOutput,
     initializeNotebookForWidgetTest
 } from './standardWidgets.vscode.common.test';
+import { GlobalStateKeyToTrackIfUserConfiguredCDNAtLeastOnce } from '../../../kernels/ipywidgets/cdnWidgetScriptSourceProvider';
 
-/* eslint-disable @typescript-eslint/no-explicit-any, no-invalid-this */
-suite('Third party IPyWidget Tests', function () {
-    let api: IExtensionTestApi;
-    const disposables: IDisposable[] = [];
-    let vscodeNotebook: IVSCodeNotebook;
-    let kernelProvider: IKernelProvider;
-
-    this.timeout(120_000);
-    const widgetScriptSourcesValue: WidgetCDNs[] = ['jsdelivr.com', 'unpkg.com'];
-    // Retry at least once, because ipywidgets can be flaky (network, comms, etc).
-    this.retries(1);
-    suiteSetup(async function () {
-        traceInfo('Suite Setup VS Code Notebook - Execution');
-        this.timeout(120_000);
-        api = await initialize();
-        const config = workspace.getConfiguration('jupyter', undefined);
-        await config.update('widgetScriptSources', widgetScriptSourcesValue, ConfigurationTarget.Global);
-        vscodeNotebook = api.serviceContainer.get<IVSCodeNotebook>(IVSCodeNotebook);
-        kernelProvider = api.serviceContainer.get<IKernelProvider>(IKernelProvider);
-        const configService = api.serviceContainer.get<IConfigurationService>(IConfigurationService);
-        const settings = configService.getSettings(undefined) as ReadWrite<IJupyterSettings>;
-        settings.widgetScriptSources = widgetScriptSourcesValue;
-
-        await workAroundVSCodeNotebookStartPages();
-        await startJupyterServer();
-        await prewarmNotebooks();
-        sinon.restore();
-        traceInfo('Suite Setup (completed)');
-    });
-    // Use same notebook without starting kernel in every single test (use one for whole suite).
-    setup(async function () {
-        traceInfo(`Start Test ${this.currentTest?.title}`);
-        sinon.restore();
-        await startJupyterServer();
-        await closeNotebooks();
-        traceInfo(`Start Test (completed) ${this.currentTest?.title}`);
-        // With less real estate, the outputs might not get rendered (VS Code optimization to avoid rendering if not in viewport).
-        await commands.executeCommand('workbench.action.closePanel');
-    });
-    teardown(async function () {
-        traceInfo(`Ended Test ${this.currentTest?.title}`);
-        if (this.currentTest?.isFailed()) {
-            await captureScreenShot(this.currentTest.title);
-        }
-        await closeNotebooksAndCleanUpAfterTests(disposables);
-        traceInfo(`Ended Test (completed) ${this.currentTest?.title}`);
-    });
-    suiteTeardown(async () => closeNotebooksAndCleanUpAfterTests(disposables));
+[true, false].forEach((useCDN) => {
```

Comment:
While enabling this test in another PR i ran into the issue this PR attempts to fix.
Now we have more tests, with & without CDN on remote, web & local.